ondewo/nlu/intent.proto

```diff
@@ -940,32 +940,55 @@ message IntentSorting {
     SortingMode sorting_mode = 2;
 }
 
-message CreateTrainingPhraseRequest {
+message UpdateTrainingPhraseBatchRequest {
+    repeated Intent.TrainingPhrase training_phrases = 1;
+}
+
+message TrainingPhraseStatus {
+    oneof phrase_or_status {
+        Intent.TrainingPhrase training_phrase = 1;
+        string error_message = 2;
+    }
+}
+
+message TrainingPhraseStatusBatch{
+    repeated TrainingPhraseStatus training_phrase_statuses = 1;
+}
+
+message CreateTrainingPhraseBatchRequest {
     // Required. The name of the intent for which to create a training phrase for.
     // Needed since the name of the training phrase may not be set upon creation.
     // Format: `projects/<Project ID>/agent/intents/<Intent ID>`.
-    string intent_name = 1;
+    message CreatePhraseRequest{
+        string intent_name = 1;
 
-    // Required. The training phrase to create.
-    Intent.TrainingPhrase training_phrase = 2;
+        // Required. The training phrase to create.
+        Intent.TrainingPhrase training_phrase = 2;
 
-    // Required. The language of training phrases
-    string language_code = 3;
+        // Required. The language of training phrases
+        string language_code = 3;
+    };
+    repeated CreatePhraseRequest training_phrase_requests = 1;
 }
 
-message GetTrainingPhraseRequest {
+message GetTrainingPhraseBatchRequest {
     // Required. The name of the training phrase.
     // Format: `projects/<Project ID>/agent/intents/<Intent ID>/trainingPhrases/<Training Phrase ID>`
-    string name = 1;
+    repeated string names = 1;
 }
 
-message UpdateTrainingPhraseRequest {
-    // Required. The training phrase to update.
-    Intent.TrainingPhrase training_phrase = 1;
-}
-
-message DeleteTrainingPhraseRequest {
+message DeleteTrainingPhraseBatchRequest {
     // Required. The name of the training phrase.
     // Format: `projects/<Project ID>/agent/intents/<Intent ID>/trainingPhrases/<Training Phrase ID>`
-    string name = 1;
-}
\ No newline at end of file
+    repeated string names = 1;
+}
+
+message DeleteTrainingPhraseStatuses{
+    message DeleteTrainingPhraseStatus{
+        oneof delete_status{
+            google.protobuf.Empty successfuly_deleted = 1;
+            string error_message = 2;
+        }
+    }
+    repeated DeleteTrainingPhraseStatus delete_statuses = 1;
```

Comment:
Say, you delete 100 phrases.

You want to check if all of them were deleted properly.
What do you do?

Seems like you need to iterate 100 elements and check if the `error_message` is empty for each of them :frowning_face: .

Same applies for Update and Create

Reply:
@gaborod16 I think in general it is quite nice to have specific error_messages for each element of the batch, but then of course your argument concerning the iteration is valid..

@andrey999333 @gaborod16 what about adding an additional status field (enum) with possible values ALL_PASSED, SOME_FAILED, ALL_FAILED, or something like that?

Reply:
i added just a boolean flag `has_errors`, so `SOME_FAILED`, `ALL_FAILED` will result into `True`. Most of the cases it will be successful, so i don't think it makes sense to refine it too much